app/src/main/kotlin/io/homeassistant/companion/android/widgets/BaseGlanceEntityWidgetReceiver.kt

```diff
@@ -131,6 +135,19 @@ abstract class BaseGlanceEntityWidgetReceiver<DAO : WidgetDao> @VisibleForTestin
                     glanceAppWidget.updateAll(context)
                 }
             }
+            ACTION_APPWIDGET_CREATED -> {
+                if (appWidgetId == AppWidgetManager.INVALID_APPWIDGET_ID) {
+                    FailFast.fail { "Missing appWidgetId in intent to add widget in DAO" }
+                } else {
+                    widgetScope.launch {
+                        // Use deprecated function to not have to specify the class of T
+                        val entity = intent.getSerializableExtra(EXTRA_WIDGET_ENTITY) as? T
```

Comment:
Given that we do this on purpose, let's also hide the warning about it

```suggestion
                        @Suppress("DEPRECATION", "UNCHECKED_CAST")
                        val entity = intent.getSerializableExtra(EXTRA_WIDGET_ENTITY) as? T
```